File: test_data/code/sample.py
class User:
    """Represents a user in the system."""

    def __init__(self, id: int, name: str, email: str):
        self.id = id
        self.name = name
        self.email = email


class UserService:
    """Provides user-related operations."""

    def __init__(self):
        self.users = {}

    def get_user(self, id: int):
        """Retrieves a user by ID."""
        return self.users.get(id)

    def create_user(self, name: str, email: str):
        """Creates a new user."""
        id = len(self.users) + 1
        user = User(id, name, email)
        self.users[id] = user
        return user


def greet(name: str) -> str:
    """Returns a greeting message."""
    return f"Hello, {name}!"


def calculate_sum(numbers: list) -> int:
    """Calculates the sum of numbers."""
    return sum(numbers)

File: test_data/code/sample.go
package sample

// User represents a user in the system
type User struct {
	ID    int
	Name  string
	Email string
}

// UserService provides user-related operations
type UserService struct {
	users map[int]*User
}

// NewUserService creates a new UserService instance
func NewUserService() *UserService {
	return &UserService{
		users: make(map[int]*User),
	}
}

// GetUser retrieves a user by ID
func (s *UserService) GetUser(id int) (*User, bool) {
	user, exists := s.users[id]
	return user, exists
}

// CreateUser creates a new user
func (s *UserService) CreateUser(name, email string) *User {
	id := len(s.users) + 1
	user := &User{
		ID:    id,
		Name:  name,
		Email: email,
	}
	s.users[id] = user
	return user
}

// Greeter interface for greeting functionality
type Greeter interface {
	Greet(name string) string
}

// Hello is a simple greeting function
func Hello(name string) string {
	return "Hello, " + name + "!"
}
